src/angularComponent/ngcOmniboxController.js

```diff
@@ -125,16 +162,66 @@ export default class NgcOmniboxController {
     } else if (keyCode === KEY.DOWN) {
       this.highlightNext();
     } else if (keyCode === KEY.ESC) {
-      this.highlightedIndex = -1;
+      this.highlightNone();
+    } else if (isSelectKey(keyCode)) {
+      const selection = this._suggestionsUiList[this.highlightedIndex];
+      selection && this._selectItem(selection.data);
     }
   }
 
   _updateSuggestions() {
-    this._suggestionItems.length = 0;
-    this.highlightedIndex = -1;
-
-    this.source({query: this.ngModel}).then((suggestions) => {
-      this.suggestions = suggestions;
+    this._suggestionsUiList.length = 0;
+    this.highlightNone();
+
+    this.source({query: this.query}).then((suggestions) => {
+      if (suggestions) {
+        this.suggestions = Array.prototype.slice.apply(suggestions);
+        this._buildSuggestionsUiList();
+      } else {
+        this.suggestions = suggestions;
```

Comment:
I don't understand this `else` clause. You're just setting `this.suggestions` to whatever falsy value the returned suggestions are? Why?